File: src/main/java/recipeapplication/repository/RecentlyViewedRepository.java
package recipeapplication.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import recipeapplication.model.RecentlyViewed;
import recipeapplication.model.User;

import java.util.List;

public interface RecentlyViewedRepository extends JpaRepository<RecentlyViewed, Long>
{
    List<RecentlyViewed> findTop5ByUserOrderByIdDesc(User user);
}

File: src/main/kotlin/recipeapplication/repository/RecentlyViewedRepository.kt
package recipeapplication.repository

import org.springframework.data.jpa.repository.JpaRepository
import recipeapplication.model.RecentlyViewed
import recipeapplication.model.User

interface RecentlyViewedRepository : JpaRepository<RecentlyViewed, Long?> {
    fun findTop5ByUserOrderByIdDesc(user: User): List<RecentlyViewed>
}